### ISSUE TITLE
Improve parallel database access management

### ISSUE THREAD

**Issue by sapetnioc** [populse/populse_db#54]
To be able to use populse_db in parallel applications, it is important to have a kind of control over locking and transaction. The current API uses SQLite DEFERRED transactions under the hood. It means that read access can be node in parallel and the first write access will switch to an EXCLUSIVE transaction (no one else can use the database). Therefore, it is not possible to simply read and modify a value atomically. It is possible to extend the API to make this possible and to also be able to modify the timeout before a "database is locked" exception is raised. Here is how an exclusive session with a custom timeout of 5 seconds could be done:

```
from populse_db import Database

database = Database('/tmp/test.sqlite', timeout=5000)
with database.exclusive as session:
    # No other process or thread can read or write
    # /tmp/test.sqlite via SQLite until the end of
    # the with statement.
    document = session['collection']['document']
    document['counter'] += 1
    session['collection']['document'] = document
```
I would prefer to be able to lock only part of the database but AFAIK it is not possible with SQLite.
